multimap/src/main/java/org/infinispan/multimap/impl/ListBucket.java

```diff
@@ -94,4 +96,36 @@ public ListBucket<V> offer(V value, boolean first) {
 
       return new ListBucket<>(newBucket);
    }
+
+   public Optional<V> index(long index) {
+      if (index == 0) {
+         return Optional.of(values.element());
+      }
+
```

Comment:
We could check for the last element, too. Something like:

```suggestion
      if (index == values.size() - 1) {
         return Optional.of(values.getLast());
      }
```